### PR TITLE
Add CHIP

### PR DESCRIPTION
**Pull request by greg-fer**
Added user guide about CHIP.
Added info about CHIP to nRF52 UG.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

**Comment by greg-fer** [comment]
@LuDuda , @Damian-Nordic , @kkasperczyk-no , drafting this for your initial comments and feedback. If you want, you can also push to this PR and merge it into https://github.com/kkasperczyk-no/sdk-nrf/tree/chip_repository_integration , but only if Kamil's branch is rebased to include recent changes on master (including structural changes in links.txt).

File: doc/nrf/ug_chip.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+   CHIP architecture overview
+
+The CHIP application layer can be broken down into several main components, from IP framing and transport management up to Data Model structure and Application itself.
+For detailed description, see the `CHIP Protocol Overview`_ page.
+
+Integration with |NCS|
+======================
+
+The CHIP project is included in the |NCS| as one of the repositories managed with the west tool.
+However, |NCS| is also one of the platforms available in CHIP.
```
Comment (Damian-Nordic) [inline doc/nrf/ug_chip.rst:47]: How about adding: "That is, the code responsible for |NCS| and CHIP integration lies in the CHIP repository" or something similar?

**Review by Damian-Nordic — COMMENTED** [review]
Looking good, just a few comments/suggestions from me.

File: doc/nrf/ug_chip.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+The CHIP protocol implementation in |NCS| supports the following features:
+
+* Data Model layer based on ZCL (Zigbee Cluster Library).
+* CHIP commissioning procedure over Bluetooth LE, SoftAP, or Ethernet.
+  Operational certificates are used for establishing secure point-to-point communication.
+
+  .. note::
+        Commissioning over Thread is not supported.
+
+* CHIP Reliable Message Protocol (CRMP) - reliable UDP-based transport protocol.
```
Comment (LuDuda) [inline doc/nrf/ug_chip.rst:27]:
```suggestion
Connected Home over IP is in the early development stage. For this reason, the list of the protocol core supported features is not yet finalized nor implemented.
```

**Review by LuDuda — CHANGES_REQUESTED** [review]
Thank you :+1: As discussed F2F please merge it with PR with initial CHIP support.

File: doc/nrf/ug_chip.rst [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+The |NCS| allows you to develop applications that are compatible with the Connected Home over IP project.
+
+Supported features
+******************
+
+The CHIP protocol implementation in |NCS| supports the following features:
+
+* Data Model layer based on ZCL (Zigbee Cluster Library).
+* CHIP commissioning procedure over Bluetooth LE, SoftAP, or Ethernet.
+  Operational certificates are used for establishing secure point-to-point communication.
```
[the remark this answers is not in the document]
Comment (greg-fer) [reply doc/nrf/ug_chip.rst:22]: Skipped as per conversation with Lukasz Duda.

File: doc/nrf/ug_chip.rst [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+The |NCS| allows you to develop applications that are compatible with the Connected Home over IP project.
+
+Supported features
+******************
+
+The CHIP protocol implementation in |NCS| supports the following features:
+
+* Data Model layer based on ZCL (Zigbee Cluster Library).
+* CHIP commissioning procedure over Bluetooth LE, SoftAP, or Ethernet.
```
[the remark this answers is not in the document]
Comment (greg-fer) [reply doc/nrf/ug_chip.rst:21]: Skipped as per conversation with Lukasz Duda.

[the remark this answers is not in the document]
Comment (greg-fer) [reply doc/nrf/ug_chip.rst:21]: Skipped as per conversation with Lukasz Duda.

**Comment by greg-fer** [comment]
Going to be merged as part of: https://github.com/kkasperczyk-no/sdk-nrf/pull/3

**Comment by greg-fer** [comment]
Closing, as this PR content's are now included in #3670

**Pull request closed without merging**